kpm/lib/kpm/account.rb

```diff
@@ -0,0 +1,386 @@
+              row[PLUGIN_NAME_COLUMN] = SAFE_PAYMENT_METHOD
+
+              if not safe_payment_method_found
+
+                safe_payment_method_found = true
+                ok_to_push = true
+              else
+                of_to_push = false
+              end
+            else
+              ok_to_push = true
+            end
+
+            if ok_to_push
+              rows.push(row)
+            end
+
+          else
+            error_importing_data = true
+            break
+          end
+        end
+
+        if tables.empty?
+          error_importing_data = true
+        end
+
+        if not ( table_name.nil? || error_importing_data )
+          tables[table_name] = rows;
+        end
+
+      end
+
+      if not error_importing_data
+        import(tables)
+      else
+        puts "\e[91;1mData on #{source_file} is invalid\e[0m\n\n"
+      end
+
+    end
+
+    def import(tables)
+      statements = Database.generate_insert_statement(tables)
+      response = nil
+
+      statements.each_key do |table_name|
+        statements_completed = 0
+        statements[table_name].each do |statement|
+          response = Database.execute_insert_statement(table_name,statement)
+
+          if response === false
+            break
+          else
+            statements_completed += 1
+          end
+        end
+
+        puts "\e[32m#{statements_completed} inserts completed of #{statements[table_name].size}\e[0m\n\n"
+
+        if response === false
+          break
+        end
+      end
+
+
+    end
+
+    def sanitize(table_name,column_name,value)
+      sanitized_value = replace_boolean(value)
+      sanitized_value = fill_empty_column(sanitized_value)
+
+      if not @tenant_record_id.nil?
+        sanitized_value = replace_tenant_record_id(column_name,sanitized_value)
+      end
+
+      if not @account_record_id.nil?
+        sanitized_value = replace_account_record_id(table_name,column_name,sanitized_value)
+      end
+
+      sanitized_value
+    end
+
+    def replace_tenant_record_id(column_name,value)
+      if column_name == 'tenant_record_id'
+        return @tenant_record_id
+      end
+
+      value
+    end
+
+    def replace_account_record_id(table_name,column_name,value)
+
+      if column_name == 'account_record_id'
+
+        return @account_record_id
+      end
+
+      if column_name == 'record_id'
+
+        if table_name == 'accounts'
+          return @account_record_id
+        else
+          return nil
+        end
+      end
+
+      if column_name == 'target_record_id'
+
+        if table_name == 'account_history'
+          return @account_record_id
+        end
+      end
+
+      value
+
+    end
+
+    def replace_boolean(value)
+      if value.to_s === 'true'
+        return 1
+      elsif value.to_s === 'false'
+        return 0
+      else
+        return value
+      end
+    end
+
+    def fill_empty_column(value)
+      if value.to_s.strip.empty?
```

Comment:
I have the feeling that we [used to insert](https://github.com/killbill/killbill/blob/master/bin/import-account#L52) `\N` in this case but maybe this is not required anymore because we are importing things differently?